File: App/Simulation.py

#   Simülasyon işlemlerinin yönetimi
#   1 -> Deprem, 2->Sel, 3-> Heyelan, 4-> Yangın
#   Simülasyon oluşturan sınıf

import random
from Earthquake import Earthquake


def earthquake():
    Earthquake(1).start()


def flood():
    print("sel")


def landslide():
    print("heyelan")


def fire():
    print("yangın")


def switchSimulation(a):
    switcher = {
        1: earthquake,
        2: flood,
        3: landslide,
        4: fire,
    }
    simulationFunction = switcher.get(a, "Invalid simulation")
    return simulationFunction()



simulationNumber = random.randint(1, 4)
switchSimulation(1)

File: App/static/Simulation.js

//  Simülasyon işlemlerinin yönetimi
//  0 -> Deprem, 1->Sel, 2-> Heyelan, 3-> Yangın
//  Simülasyona ait bilgilerin toplanıp server'a yollanacağı sınıf

var simulationNumber = Math.floor(Math.random() * 4);

switch (simulationNumber) {
    case 0:
        earthquake();
        break;
    case 1:

        break;
    case 2:

        break;
    case 3:

        break;

}

function createSimulation(disaster) {
    showEarthquakeEffect();
    disaster.start();

}

function showEarthquakeEffect() {
    var element = document.getElementById("map");
    element.classList.add("shake-opacity");
    element.classList.add("shake-constant");
    setTimeout(function () {
        element.classList.remove("shake-opacity");
        element.classList.remove("shake-opacity");
    }, 1500);
    
}

